### PR TITLE
Pass through specific github label to presumit builds

### PR DESCRIPTION
**Pull request by yusuf-goog**
Bug:https://github.com/flutter/flutter/issues/130076

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/service/luci_build_service.dart [hunk 1]
```diff
@@ -188,6 +188,7 @@ class LuciBuildService {
         'check_run_id': checkRun.id,
         'commit_sha': sha,
         'commit_branch': pullRequest.base!.ref!.replaceAll('refs/heads/', ''),
+        'github_build_label': pullRequest.labels!.any((label) => label.name == 'github_build_label'),
```
Comment (godofredoc) [inline app_dart/lib/src/service/luci_build_service.dart:191]: Should we add tests?

File: app_dart/lib/src/service/luci_build_service.dart [hunk 2]
```diff
@@ -196,6 +196,7 @@ class LuciBuildService {
 
       final Map<String, Object> properties = target.getProperties();
       properties.putIfAbsent('git_branch', () => pullRequest.base!.ref!.replaceAll('refs/heads/', ''));
+      properties['github_build_label'] = pullRequest.labels!.any((label) => label.name == 'github_build_label');
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:199]: Nit: use a variable instead and add a doc explaining what this means.

File: app_dart/lib/src/service/luci_build_service.dart (hunk 1 above)
Comment (yusuf-goog) [reply app_dart/lib/src/service/luci_build_service.dart]: done

File: app_dart/lib/src/service/luci_build_service.dart (hunk 2 above)
Comment (yusuf-goog) [reply app_dart/lib/src/service/luci_build_service.dart]: done

File: app_dart/lib/src/service/luci_build_service.dart [hunk 3]
```diff
@@ -62,6 +62,10 @@ class LuciBuildService {
   static const int kDefaultPriority = 30;
   static const int kRerunPriority = 29;
 
+  /// Github labels have a max length of 100, so conserve chars here.
+  static const String githubBuildLabelPrefix = 'gh_override:';
+  static const String propertiesGithubBuildLabelName = 'gh_overrides';
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:67]: Nit: `gh_override_labels`.

File: app_dart/lib/src/service/luci_build_service.dart [hunk 4]
```diff
@@ -62,6 +62,10 @@ class LuciBuildService {
   static const int kDefaultPriority = 30;
   static const int kRerunPriority = 29;
 
+  /// Github labels have a max length of 100, so conserve chars here.
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:65]: Nit: doc that these are used for packages repo only, and nice to add a link pointing to the pacakges recipes.

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: app_dart/lib/src/service/luci_build_service.dart [hunk 5]
```diff
@@ -62,6 +62,12 @@ class LuciBuildService {
...
+  /// This is currently used by packages repo only.
+  /// See: https://github.com/flutter/flutter/issues/130076
+  static const String githubBuildLabelPrefix = 'gh_override:';
```
Comment (stuartmorgan) [inline app_dart/lib/src/service/luci_build_service.dart:68]: Why do we need a `gh_` prefix on GitHub labels? What is `gh_` distinguishing from in the context of GitHub label names?

Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:68]: If I recall yusuf is porting a functionality that existed in cirrus? What was this defined as previously? Or was it?

**Review by ricardoamador — APPROVED** [review]
LGTM

File: app_dart/lib/src/service/luci_build_service.dart (hunk 5 above)
Comment (stuartmorgan) [reply app_dart/lib/src/service/luci_build_service.dart]: Cirrus puts all PR labels in an environment variable called `CIRRUS_PR_LABELS`. There is no magic prefix in the Cirrus version of the functionality.

**Pull request merged**